src/AutoMapper/PropertyMap.cs

```diff
@@ -37,7 +39,7 @@ public PropertyMap(PropertyMap inheritedMappedProperty, TypeMap typeMap)
 
         public Type DestinationPropertyType => DestinationProperty.GetMemberType();
 
-        public IEnumerable<MemberInfo> SourceMembers => _memberChain;
+        public IList<MemberInfo> SourceMembers => _memberChain;

```

Comment:
I'm checking the count. If it's greater than one, I assume it's the default flattening.